TKeazirian.HTTPServer/EchoServer.cs

```diff
@@ -6,61 +6,80 @@ namespace TKeazirian.HTTPServer;
 
 public static class EchoServer
 {
-    private static string? request;
+    private static string? _request;
 
     public static void StartListening()
     {
-        IPHostEntry ipHostInfo = Dns.GetHostEntry(Dns.GetHostName());
-        IPAddress ipAddress = IPAddress.Parse("127.0.0.1");
-        IPEndPoint localEndPoint = new IPEndPoint(ipAddress, 11000);
-
-        Socket listener = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+        var listener = CreateSocketListener(out var localEndPoint);
```

Comment:
I'm not a fan of out parameters.  maybe break that line out.   output parameters are allowed but unexpected behavior.